### PR TITLE
Modify Edit Command to align with View Command

### PR DESCRIPTION
**Pull request by Gavino3o**
# Overview 
Tackles #63 

## Changes
- `edit` command now edits the person in the view panel only. 
- `edit` command now does not require a `PersonType` tag. The parser will check the `PersonType` of the person being viewed and handle accordingly.
- Fix a bug where locations for specialists were not being considered when checking if a person is being edited.
- Added and modified some new error messages for `edit` and `find` command.

## Notes:
I have added/modified relevant test cases for the new `edit` command. Code Cov a bit strict :(

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -131,6 +130,33 @@ Examples:
 * `find -sp n/Alex s/Orthopaedic` returns any specialists names including the string `Alex` who has the `Orthopaedic` specialty
 <br>
 
+### Editing a pre-existing patient or specialist record: `edit`
```
Comment (vansh284) [inline docs/UserGuide.md:133]: Great update of the user guide! Would be good if you could update the command summary table at the bottom as well.

**Review by vansh284 — COMMENTED** [review]
Nice implementation! Seems to be a big change so would be great if others could take a look as well.

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 2]
```diff
@@ -119,6 +129,14 @@ public Command parseCommand(String userInput) throws ParseException {
...
 
             switch (commandWord) {
+            case EditCommand.COMMAND_WORD:
+                if (model.getSelectedPerson() instanceof Patient) {
+                    return new EditCommandParser().parse(PersonType.PATIENT, arguments);
+                }
+                if (model.getSelectedPerson() instanceof Specialist) {
+                    return new EditCommandParser().parse(PersonType.SPECIALIST, arguments);
+                }
+                break;
```
Comment (alvinlim277) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:139]: Nice way of getting the type of viewed persons into the `EditCommandParser`.

**Review by alvinlim277 — COMMENTED** [review]
LGTM overall! Will be good if you can improve the code coverage as well, the remaining uncovered lines seem to be related to error messages.

File: docs/UserGuide.md (hunk 1 above)
Comment (Gavino3o) [reply docs/UserGuide.md:133]: Thanks for pointing that out. I added the edit command into the command summary table.

**Review by alvinlim277 — APPROVED** [review]
LGTM

**Pull request merged**